src/components/NftRewards/NftRewardsSection.tsx

```diff
@@ -172,9 +223,9 @@ export function NftRewardsSection() {
                   rewardTierUpperLimit={
                     rewardTiers?.[idx + 1]?.contributionFloor
                   }
-                  isSelected={idx === selectedIndex}
-                  onClick={() => handleSelected(rewardTier, idx)}
-                  onRemove={deselectTier}
+                  isSelected={selectedIndices.includes(idx)}
```

Comment:
**question (non-blocking)** I don't really understand why we dont use the tier ID instead of the index. Tier ID seems safer and less messy overall. WDYT?

Reply:
Yeah I agree, I've overhauled the unnecessary indice logic